### PR TITLE
Major refactoring

### PR DESCRIPTION
**Pull request by xrotwang**
- dropping py2 compat and removing obsolete code
- encapsulating xml building in a utility module, providing
  - better IDE support (since each valid xml tag is a function)
  - simpler element creation:
    - text content is a keyword argument,
    - attribute values are cast to string depending on context

### CODE REVIEW

**Comment by xrotwang** [comment]
I admit that this is rather big and should not be the norm for PRs. I also made the mistake of bundling py2 removal with xml creation refactoring. But we have decent test coverage by now, so I'm fairly convinced that I didn't break anything. I also like the xml creation now, in particular because turning xml tags into python functions makes such a difference when using an IDE like PyCharm.

File: setup.py [hunk 1]
```diff
@@ -22,10 +17,11 @@
         'Programming Language :: Python :: 3.7',
         'License :: OSI Approved :: BSD License',
     ],
-    packages=['beastling', 'beastling.clocks', 'beastling.fileio', 'beastling.models', 'beastling.treepriors'],
+    packages=find_packages(),
```
Comment (lmaurits) [inline setup.py:20]: Oh, wonderful!  I always thought it was seriously broken that we had to specify these manually one by one.

**Review by lmaurits — APPROVED** [review]
This definitely looks like an improvement to me.

It's a good idea to make it as easy as possible for people who author interesting new BEAST classes (like tree priors) to also write BEASTling classes for it.  So getting this kind of much neater and simpler code for functionality is IMHO well worth the cost of the quite low complexity of something like `utils/xml.py`.

@Anaphory, what do you think?

**Comment by xrotwang** [comment]
It's low complexity now, but once the xml creation goes through this
central point, one could do more. Switch to a different xml backend (if
needed), validate input, e.g. plates always must have var and range, ...

Luke Maurits <notifications@github.com> schrieb am Do., 1. Aug. 2019, 22:23:

**Review by Anaphory — APPROVED** [review]
These are good changes in general.

One thing is important to keep in mind: In our current way of describing, we are using the names of the inputs of a parent class as the tag, specifying the class in the `spec=` attribute. The approach to use the class name of the input as tag and in it specifying the parent's input as `name=` is a bit more consistent with having tag functions which might at some point obtain checks and balances on their attributes.

File: setup.py [hunk 2]
```diff
@@ -22,10 +17,11 @@
...
     install_requires=[
         'newick>=0.6.0',
         'appdirs',
+        'csvw',
         'clldutils>=2.8',
```
Comment (Anaphory) [inline setup.py:25]: Shouldn't the clldutils dependency be removed now?

**Comment by xrotwang** [comment]
Not yet. But the way is paved now. Waiting for this to be merged 😀

Gereon Kaiping <notifications@github.com> schrieb am Di., 6. Aug. 2019,
14:24:

**Comment by lmaurits** [comment]
Sorry for the delay in merging this, I didn't want to do it until @Anaphory had had a chance to look over it and give his approval.  Happy it's done now!

**Pull request merged**